### ISSUE TITLE
Kafka does not not work with different k6 load profiles

### ISSUE THREAD

**Issue by luminorena** [mostafa/xk6-kafka#223]
Hi, @mostafa

I have encontered with lots of errors trying to use your library to load Kafka in production. I use these two scripts. 
This is the script which uses your xk6-kafka-library (I have built the last version):

```
import { check } from "k6";
import {
  Writer,
  Reader,
  Connection,
  SchemaRegistry,
  CODEC_SNAPPY,
  SCHEMA_TYPE_JSON,
} from "k6/x/kafka";

import { LoadJKS, TLS_1_2 } from "k6/x/kafka";
const jks = LoadJKS({
  path: "kafka.keystore.jks",
  password: "password",
  clientCertAlias: "localhost",
  clientKeyAlias: "localhost",
  clientKeyPassword: "password",
  serverCaAlias: "localhost",
  });

const tlsConfig = {
  enableTls: true,
  insecureSkipTlsVerify: false,
  minVersion: TLS_1_2,
  clientCertPem: jks["clientCertsPem"][0],
  clientKeyPem: jks["clientKeyPem"],
  serverCaPem: jks["serverCaPem"],
};

const brokers = ["localhost:9092"];
const topic = "test_1";

const writer = new Writer({
  brokers: brokers,
  tlsConfig,
  topic: topic,
  autoCreateTopic: true,
  compression: CODEC_SNAPPY,
});

const reader = new Reader({
  brokers: brokers,
  tlsConfig,
  topic: topic,
});

const connection = new Connection({
  address: brokers[0],
  tlsConfig,
});

const schemaRegistry = new SchemaRegistry();

if (__VU == 0) {
  connection.createTopic({
    topic: topic,
    configEntries: [
      {
        configName: "compression.type",
        configValue: CODEC_SNAPPY,
      },
    ],
  });
}

export const options = {
  thresholds: {
    kafka_writer_error_count: ["count == 0"],
    kafka_reader_error_count: ["count == 0"],
  },
};

function getRandomInt(min, max) {
  return min + Math.floor(Math.random() * (max - min + 1));
}

export default function () {
  let messages = [
    {
      key: schemaRegistry.serialize({
        data: {
          correlationId: "test-id-abc-" + getRandomInt(10, 100),
        },
        schemaType: SCHEMA_TYPE_JSON,
      }),
      value: schemaRegistry.serialize({
        data: {
          menu: {
            version: "test_1",
            author: "test_2",
            description: "test_3",
          }

        },
        schemaType: SCHEMA_TYPE_JSON,
      })

    }
  ];

  writer.produce({ messages: messages });

  messages = reader.consume({});

}

export function teardown(data) {
  console.log(Object.keys(jks));
  console.log(jks);
  console.log(tlsConfig);

  writer.close();
  reader.close();
  connection.close();
}

```

 And this is the load profile with k6:

```
import {sleep} from 'k6'
import scenario_1 from '<path_to_scenario_1.js>'

export const options = {
scenarios: {
	scenario_1: {
	executor: 'constant-vus',
	vus: 1,
	duration: '30s'
		}
	}
}

export function all(){
    scenario_1();
}

export default function (setupData) {
all();
sleep(3);
}
```
The problem is when I run just the first script I have no errors and one message is written to Kafka successfully.
But when I need to use load profile with any intensity, not all messages are written to Kafka, lots of errors occur.

1. One user:
```
level=error msg="Error writing messages., OriginalError: %!w(kafka.writeErrors=[some different hex numbers])" error="Error writing messages., OriginalError: %!w(kafka.writeErrors=[some different hex numbers])" ted iterations
at github.com/mostafa/xk6-kafka.(*Kafka).writerClass.func1 (native)
at writer.produce({messages: messages}) - this is the line in the script where I use your library
level=error msg="Unable to read messages."
```
By the way, do we need to read here something if we just write to Kafka? Or do I misunderstand something ?

2. When I use profile with more then 1 vu (10 or more)  I can also get these errors (in addition to ones mentioned above)

```
level=error msg="Error writing messages., OriginalError: %!w(*tls.CertificateVerificationError=${[different hexs] {hex, <nil>, <nil>}}

level=error msg="Error writing messages., OriginalError: %!w(*fmt.wrapError=&{unexpected EOF: broker appears to be expecting TLS hex number})" (repeats the times)
```
3. Also, I do not know why but sometimes the initialization fails. I can run script and this error occurs. If I rerun it, I goes ok.
```
Init  ======> 17/20 Vus initialized
GoError: Failed to get controller., Original error: %!w(*errors.errorString=&{EOF})
```
6. In addition, I cannot use two load scenarios with one certificate. I have two scripts which use the same jks certificate and in my load profile I put them like this:

```
import {sleep} from 'k6'
import scenario_1 from '<path_to_scenario_1.js>'
import scenario_2 from '<path_to_scenario_2.js>'

export const options = {
scenarios: {
	scenario_1: {
	executor: 'constant-vus',
	vus: 1,
	duration: '30s'
		}
	},
	scenario_2: {
	executor: 'constant-vus',
	vus: 1,
	duration: '30s'
		}
	}
}

export function all(){
    scenario_1();
    sleep(1);
    scenario_2();
}

export default function (setupData) {
all();
sleep(3);
}
```

If I run the test, I will give the error:
```
Failed to write cert file, OriginalError: %!w(*fs.PathError=&{open client-cert-2.pem 1224})
at github.com/mostafa/xk6-kafka.(*kafka).LoadJKSFunction-fm (native)
at const jks = LoadJKS line (script with your library)
```
I think this is because you create pem certs in root directory and two scripts use it at the same time. I have created two directories and put certs for each Kafka scripts there (it is the same cert, I have just copied it there). And your go code uses the certs in root directory. 

Am I doing something wrong?

**Comment by mostafa** [comment]
Hey @luminorena,

xk6-kafka uses kafka-go internally, which means that it follows the event loop logic of the kafka-go library, not that of the k6. k6 uses these load profiles to simulate HTTP traffic, and I think they can't be used for other uses cases.

1. I have demonstrated that normal load profiles can be used and scaled up to a few tens of VUs in the [README](https://github.com/mostafa/xk6-kafka#k6-test-scripts). And, you _don't have to_ use a reader to consume any messages from Kafka. You can just write messages and let your system/platform/backend process them.
2. I fixed a few bugs in the `LoadJKS` function that were released in [v0.18.2](https://github.com/mostafa/xk6-kafka/releases/tag/v0.18.2), yet I released [v0.19.0](https://github.com/mostafa/xk6-kafka/releases/tag/v0.19.0) that contains updates to Go, k6 and dependencies. Please try the latest version instead. All in all, it seems like a TLS config issue.
3. Not exactly sure why, but it might be due to cluster leader election or something similar. If you run a cluster locally for testing, it might be due to initialization issues.
4. You shouldn't have any issue with that, unless you're running `LoadJKS` multiple times. Run it once in init code and just reuse the constant.

**Comment by mostafa** [comment]
@luminorena 

I suppose this is resolved. Feel free to reopen it if the issue persists.
